doc/groups.md

```diff
@@ -1 +1,315 @@
 # Groups Module
+Module | HTTP Method | URL | Description 
+--- | --- | --- | ---
+[Add Groups](#add) | POST | /groups-add | Add Groups API
+[Edit Groups](#edit) | PUT | /groups-edit/{id} | Edit Groups API
+[View All Groups](#view) | GET | /groups | View Groups API
+[Delete Groups](#delete) | DELETE | /groups-delete/{id} | Delete Groups API
+
+
+### Database
+![](./groups-layer.png)
+
+*For [Add Groups](#add) and [Edit Groups](#edit), you need get aros to set aros_id.* 
+
+## <a name="add"></a>Add Groups
+
+### Endpoint 
+POST /groups-add
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type | application/json
+Accept | application/json
+
+### Request Payloads
+Name | Type | Example Value
+--- | --- | ---
+title | string | Admin  
+description | string | Administrator
+aros_id | int | 1
+```
+{
+    "title": "Admin",
+    "description": "Administrator",
+    "aros": {
+      "aros_id": "1"
+    }
+}
+
+```
+
+### Response Payloads
+HTTP Code | Status | Description
+--- | --- | ---
+400 | Bad Request | Bad request payload  
+404 | Not Found | User not found in database  
+500 | Internal Server Error | some un-handle error in server 
+201 | Created | Created
+```
+{
+    "status_code": "CDC-400",
+    "status_message": "Bad Request",
+    "data": null
+}
+```
+
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": {
+        "id": "1",
+        "title": "Admin",
+        "description": "Administrator",
+        "created": "2020-12-01 00:00:00",
+        "modified": "2020-12-01 00:00:00",
+        "aros_id": "1"
+    }
+} 
+```
+
+### Logic
+
+#### Validation
+- title : required and not empty
+- description : required and not empty
+- aros_id : required and not empty
```

Comment:
remove aros_id. after create new group, you need insert new aros for current group.